packages/flutter/lib/src/painting/fractional_offset.dart

```diff
@@ -113,22 +117,22 @@ abstract class FractionalOffsetGeometry {
     if ((a == null || a is FractionalOffsetDirectional) && (b == null || b is FractionalOffsetDirectional))
       return FractionalOffsetDirectional.lerp(a, b, t);
     if (a == null) {
-      return new _MixedFractionalOffset(
-        ui.lerpDouble(0.5, b._dx, t),
-        ui.lerpDouble(0.0, b._start, t),
+      return new _SchrodingersFractionalOffset(
```

Comment:
Can we lerp from topStart?

Reply:
I think we can (and should, for consistency with regular FractionalOffset and Offset). _dxForRTL would lerp from 1.0 and _dxForLTR would lerp from 0.0.

Reply:
I think we should also revert my change to FractionalOffset.lerp. It should lerp from 0,0. It's just the same as lerping an Offset, it's a vector, it should lerp from magnitude 0.

Reply:
Ok.  I'd like to do that in a separate patch, if that's ok, since it's a separate change to the semantics (e.g., we might decide later to revert it or something).

Reply:
Yeah, that's fine